highfidelity/hifi
# Minimum entity dimension (1mm)

Fixes several bugs:
- Primitives with any dimension <= 0 render as 1x1x1
- Models with any dimension == 0 trigger an assert in debug
- EntityItems with sufficiently small dimensions disappear from the Octree completely because they fail the  `AACube::contains(const AACube& otherCube)` check in `UpdateEntityOperator::subTreeContainsNewEntity` (before they hit the `SMALLEST_REASONABLE_OCTREE_ELEMENT_SCALE` in `EntityTreeElement::bestFitBounds`) due to floating point imprecision.

Test plan:
- Run [this](https://gist.githubusercontent.com/SamGondelman/7165900b9500732ebf77cd4884427a28/raw/6e4acee544d9aa5eb691d16a49d22bbbe0e99c5a/ZeroDimensions.js) in an empty domain.
- You should see 6 entities: 2 shrinking and growing uniformly, 2 vertically, 2 non-oscillating and flat.
- Open Create.  You should see 8 entities in the entity list (there's also 2 more on the right that are just very small).
 - On Android, go to the same domain and look at the entities.  You shouldn't crash.
